torch/storage.py

```diff
@@ -1493,7 +1493,7 @@ class _LegacyStorageMeta(type):
     dtype: torch.dtype
 
     def __instancecheck__(cls, instance):
-        if type(instance) == TypedStorage:
+        if isinstance(instance, TypedStorage):
```

Comment:
So the other changes are more likely to be find since we override `__instancecheck__` here to explicitly forbid submodules.... Not sure if this is valid